Mage/src/main/java/mage/abilities/effects/keyword/RadCounterTriggeredAbility.java

```diff
@@ -0,0 +1,89 @@
+package mage.abilities.effects.keyword;
+
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Cards;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+
+/**
+ * 725.1. Rad counters are a kind of counter a player can have (see rule 122, "Counters").
+ * There is an inherent triggered ability associated with rad counters. This ability has no
+ * source and is controlled by the active player. This is an exception to rule 113.8. The
+ * full text of this ability is "At the beginning of each player's precombat main phase, if
+ * that player has one or more rad counters, that player mills a number of cards equal to
+ * the number of rad counters they have. For each nonland card milled this way, that player
+ * loses 1 life and removes one rad counter from themselves."
+ *
+ * @author Susucr
+ */
+public class RadCounterTriggeredAbility extends TriggeredAbilityImpl {
+    public RadCounterTriggeredAbility() {
+        super(Zone.ALL, new RadCounterEffect());
+        setTriggerPhrase("At the beginning of your precombat main phase, if you have one or more rad counters, ");
```

Comment:
should this be set up as an intervening if? I guess it probably doesn't matter because nothing happens if no rad counters at resolution, but just to be as consistent as possible with the rules.